addons/hr_evaluation/hr_evaluation.py

```diff
@@ -1,375 +1,465 @@
+    def _check_employee_appraisal_duplication(self):
+        """ Avoid duplication"""
+        if self.employee_id and self.department_id and self.date_close:
+            date_closed = datetime.datetime.strptime(self.date_close, DTF)
+            start_datetime = time.strftime(str(date_closed.year) + '-' + str(date_closed.month) + '-01' + ' 00:00:00')
+            end_datetime = time.strftime(str(date_closed.year) + '-' + str(date_closed.month) + '-' + str(date_closed.day) + ' 23:59:59')
+            appraisal_ids = self.search([
+                ('employee_id', '=', self.employee_id.id), ('department_id', '=', self.department_id.id),
+                ('date_close', '<=', end_datetime),
+                ('date_close', '>=', start_datetime)])
+            if len(appraisal_ids) > 1:
+                raise UserError(_("You cannot create more than one appraisal for same Month & Year"))
+
+    @api.model
+    def create_message_subscribe_users(self, subscribe_users):
+        user_ids = [emp.user_id.id for emp in subscribe_users if emp.user_id]
+        if self.employee_id.user_id:
+            user_ids.append(self.employee_id.user_id.id)
+        if self.employee_id.department_id.manager_id.user_id:
+            user_ids.append(self.employee_id.department_id.manager_id.user_id.id)
+        if self.employee_id.parent_id.user_id:
+            user_ids.append(self.employee_id.parent_id.user_id.id)
+        return self.message_subscribe_users(user_ids=user_ids)
+
+    @api.model
+    def create(self, vals):
+        res = super(hr_evaluation, self.with_context(mail_create_nolog=True)).create(vals)
+        if res.appraisal_manager_ids:
+            res.create_message_subscribe_users(res.appraisal_manager_ids)
+            res.message_post(body=_("Employee Appraisal created"), subtype="mail.mt_comment", type="notification")
         return res
 
-    def survey_req_waiting_answer(self, cr, uid, ids, context=None):
-        request_obj = self.pool.get('survey.user_input')
-        for interview in self.browse(cr, uid, ids, context=context):
-            if interview.request_id:
-                request_obj.action_survey_resent(cr, uid, [interview.request_id.id], context=context)
-            self.write(cr, uid, interview.id, {'state': 'waiting_answer'}, context=context)
+    @api.multi
+    def write(self, vals):
+        emp_obj = self.env['hr.employee']
+        for evl_rec in self:
+            if self.state == 'new' and vals.get('state') == 'done':
+                raise UserError(_("""Sorry ! You cannot drag this card from the "To Start" column to the "Done" column. You have to drag it to the "Appraisal Sent" column first."""))
+            # avoid recursive process
+            if vals.get('state') == 'pending' and not evl_rec._context.get('send_mail_status'):
+                evl_rec.button_send_appraisal()
+            if vals.get('interview_deadline') and self.state == 'pending' and not vals.get('meeting_id'):
```

Comment:
Why not ? If I want to create a record for an interview that has already been done ... your constraint does not seem necessary to me. Like said above, do not put hard constraints, ease the user flow instead.